File: src/test/java/cn/edu/gxust/jiweihuang/java/math/fitter/LogisticFitterTest.java
package cn.edu.gxust.jiweihuang.java.math.fitter;

import cn.edu.gxust.jiweihuang.java.math.function.univariate.Logistic;
import org.hipparchus.fitting.WeightedObservedPoint;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class LogisticFitterTest {
    public static void main(String... args) {
        Logistic logistic = new Logistic(0.8, 0.9, 0.1);
        double[] x = new double[]{-0.5,0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6};
        List<WeightedObservedPoint> observedPoints = new ArrayList(x.length);
        for (double xi : x) {
            System.out.println(xi + ":" + logistic.value(xi));
            observedPoints.add(new WeightedObservedPoint(0, xi, logistic.value(xi)));
        }
        System.out.println(Arrays.toString(LogisticFitter.create().withStartPoint(new double[]{1, 1, 1}).fit(observedPoints)));
    }
}

File: src/test/scala/cn/edu/gxust/jiweihuang/scala/math/fitter/LogisticFitterTest.scala
package cn.edu.gxust.jiweihuang.scala.math.fitter

import java.util

import cn.edu.gxust.jiweihuang.scala.math.function.univariate.Logistic
import cn.edu.gxust.jiweihuang.scala.test.UnitSpec
import org.hipparchus.fitting.WeightedObservedPoint

class LogisticFitterTest extends UnitSpec {
  "LogisticFitter" should "fit Logistic Function" in {
    val logistic: Logistic = Logistic(-1, 6, 3)
    println(s"Logistic(-1,2,3) = ${logistic}")
    val xData = 1 to 100
    println(s"{1 to 20} = ${xData}")
    val yData = for (x <- xData) yield logistic.value(x)
    println(util.Arrays.toString(yData.toArray))

    val observations = new util.ArrayList[WeightedObservedPoint](xData.length)
    for (x <- xData) {
      observations.add(new WeightedObservedPoint(0, x, logistic.value(x)))
    }
    val logisticFitter = LogisticFitter(Array[Double](0, 0, 0))
    val parameter: Array[Double] = logisticFitter.fit(observations)
    println(util.Arrays.toString(parameter))
    val logisticf = logisticFitter.fitToLogistic(observations)
    println(logisticf)
  }
}
